libyul/optimiser/SSATransform.cpp

```diff
@@ -135,6 +183,33 @@ void SSATransform::operator()(Block& _block)
 				boost::get<VariableDeclaration>(statements.front()).variables = std::move(newVariables);
 				return { std::move(statements) };
 			}
+			else if (
+				_s.type() == typeid(If) ||
+				_s.type() == typeid(Switch) ||
+				_s.type() == typeid(ForLoop) ||
+				_s.type() == typeid(Block)
+			)
+			{
+				auto variableValuesBefore = m_currentVariableValues;
+				visit(_s);
+				vector<Statement> statements;
+				auto loc = locationOf(_s);
+				statements.emplace_back(std::move(_s));
+				for (YulString varName: variableValuesBefore | boost::adaptors::map_keys)
+				{
+					if (!m_currentVariableValues.count(varName))
```

Comment:
Can you explain why this is the right condition?

Reply:
Comments added

Reply:
Ok, so this basically does what `Assignments(_s)` does in other parts of the code, right?

Reply:
Yes